### DISCUSSION TITLE
PPOTrainer for models

### DISCUSSION THREAD

**Discussion by JarcauCristian** [h2oai/h2o-llmstudio#398] [Q&A]
Does llmstudio train using the PPOTrainer from the trl package, or it can be done afterwards somehow?

**Comment by pascal-pfeiffer** [comment]
H2O LLM Studio uses a modified version of the trl trainer in the background when using the RLHF problem type
![image](https://github.com/h2oai/h2o-llmstudio/assets/1069138/d146485d-0ae6-4069-b57d-53aabf6584a4)

https://github.com/h2oai/h2o-llmstudio/blob/bf667d03e6c47940226b234f764803604e380e87/llm_studio/src/trl/trainer.py#L1-L18

**Reply by JarcauCristian** [reply]
Alright! Thank you very much!
